tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/BuilderUtil.scala

```diff
@@ -77,6 +77,34 @@ object BuilderUtil {
     } else ret
   }
 
+  /**
+   * Some (variadic) operators introduce bound variables (e.g. map, funDef). This method constructs the expressions
+   * associated with the operator, and additionally performs shadowing checks and bound-variable tagging.
+   */
+  def boundVarIntroductionVariadic(
+      rawMethod: (TlaEx, Seq[(TlaEx, TlaEx)]) => TlaEx
+    )(ex: TBuilderInstruction,
+      varSetPairs: (TBuilderInstruction, TBuilderInstruction)*): TBuilderInstruction = for {
+    bodyEx <- ex
+    boundAfterBodyEx <- allBound // variables may not appear as bound in bodyEx
+    pairs <- varSetPairs.foldLeft(Seq.empty[(TlaEx, TlaEx)].point[TBuilderInternalState]) {
```

Comment:
From the implementation point of view, would not it be more efficient to carry a hash map of strings (variable names) to `TlaEx`?

Reply:
Maybe marginally, but from the interface point of view, it's easier to invoke `metod(a -> b, c ->d)`, since `->` makes a `(T,T)` pair.